astropy/coordinates/transformations.py

```diff
@@ -706,8 +706,8 @@ def __init__(self, fromsys, tosys, priority=1, register_graph=None):
                 hasattr(tosys, 'get_frame_attr_names')):
             # the if statement is there so that non-frame things might be usable
             # if it makes sense
-            for from_nm in fromsys.get_frame_attr_names():
-                if from_nm in tosys.get_frame_attr_names():
+            for from_nm in fromsys.frame_attributes.keys():
```

Comment:
Had to think above this, but it is definitely OK and avoids creating a needless `OrderedDict`.

Reply:
Yeah, we should probably consider deprecating `get_frame_attr_names` in favor of this usage, given that `get_frame_attr_names` actually gives a dict not just the names... I think we discussed this at one point but never implemented it.